skada/deep/_adversarial.py

```diff
@@ -421,11 +429,13 @@ def forward(
         pseudo_label_s = torch.argmax(y_pred_s, axis=-1)
         pseudo_label_t = torch.argmax(y_pred_t, axis=-1)
 
-        disc_loss_src = self.disc_criterion_(disc_pred_s, pseudo_label_s)
-        disc_loss_tgt = self.disc_criterion_(disc_pred_t, pseudo_label_t)
+        disc_criterion_s_ = torch.nn.CrossEntropyLoss()
+        disc_loss_s = disc_criterion_s_(disc_pred_s, pseudo_label_s)
+        disc_criterion_t_ = ModifiedCrossEntropyLoss()
+        disc_loss_t = disc_criterion_t_(disc_pred_t, pseudo_label_t)
```

Comment:
Is it ok to instantiate functions in the forward pass (I mean from the computational efficiency point of view)?
Maybe it is better to define the criterions in the init as attributes and then use the same in forward?